src/transformStateWithClones.js

```diff
@@ -6,8 +6,53 @@
  *
  * @return {Object[]}
  */
+function deepCopy(obj) {
+  if (obj === null) {
+    return null;
+  }
+
+  if (typeof obj !== 'object') {
+    return obj;
+  }
+
+  const copy = {};
+
+  for (const key in obj) {
+    copy[key] = deepCopy(obj[key]);
+  }
+
+  return copy;
+}
+
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  const state2 = deepCopy(state);
+
+  for (const a of actions) {
+    if (a.type === 'addProperties') {
```

Comment:
Everything is great but I think we need to use switch/case command here. It was in this (stupid) checklist in readme file:
https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md
It takes 5 minutes, hope it won't bother you much.

Reply:
I meant switch instead of all those if (a.type ===.......)
Good luck!

Reply:
Thx, added fixes. :)